opennms-webapp-rest/src/main/java/org/opennms/web/rest/v1/NodeRestService.java

```diff
@@ -123,6 +129,22 @@ public OnmsNodeList getNodes(@Context final UriInfo uriInfo) {
             builder.distinct();
 
             crit = builder.toCriteria();
+        } else if (params.getFirst("filterRule") != null) {
+            final Set<Integer> filteredNodeIds = m_filterDao.getNodeMap(params.getFirst("filterRule")).keySet();
+            if (filteredNodeIds.size() < 1) {
+                // The "in" criteria fails if the list of node ids is empty
+                final OnmsNodeList coll = new OnmsNodeList(Collections.emptyList());
+                coll.setTotalCount(0);
+                return coll;
+            }
+
+            // Apply the criteria, without the filter rule
+            params.remove("filterRule");
+            final CriteriaBuilder filterRuleCriteriaBuilder = getCriteriaBuilder(params);
+            filterRuleCriteriaBuilder.in("id", filteredNodeIds);
+            filterRuleCriteriaBuilder.distinct();
+
+            crit = filterRuleCriteriaBuilder.toCriteria();
```

Comment:
To me this looks almost like the first `nodeId` implementation.
Can we either use the same criteria as above, or at least add a comment why we filter differently (and why we need it)?